javascript/app.js

```diff
@@ -43,36 +83,73 @@ function makeRandomProductArray() {
 }
 
 let randomProducts = [];
-function randomRenderImage(){
+function randomRenderImage() {
   randomProducts = makeRandomProductArray();
-  for(let i = 0; i < randomProducts.length; i++){
+  for (let i = 0; i < randomProducts.length; i++) {
     let randomProduct = randomProducts[i];
 
     randomProduct.views++;
     randomProduct.render(i);
   }
 }
+function onClick(event) {
+  let id = event.target.id;
+  let currentClicks = 0;
+  let totalClicksAllowed = 10;
+  if (currentClicks === totalClicksAllowed) {
+    for (let i = 0; i < 2; i++) {
+      let img = document.getElementById(`image${i}`);
+      img.removeEventListener("click", clickHandler);
+    }
+    alert("Voting has ended, please View Results.");
+  } else {
+    currentClicks++;
+    randomProducts[`${id[5]}`].clicks++;
+    randomRenderImage();
+  }
+}
+
+// function renderList() {
+//   let divContainer = document.getElementById("data-list");
 
-function addClickHandler(i) {
+//   for (let i = 0; i < allProducts.length; i++) {
+//     let product = allProducts[i];
+//     let listItem = document.createElement("li");
+//     listItem.innerText = `${product.name} was clicked ${product.clicks} times, and seen ${product.views} times.`;
+//     divContainer.appendChild(listItem);
+//   }
+// }
+Product.prototype.onClick = function (i) {
   let img = document.getElementById(`image${i}`);
-  img.addEventListener("click", randomRenderImage);
-  allProducts[i].clicks++;
-}
+  img.addEventListener("click", onClick);
+  };
 
-function renderList(){
-  let divContainer = document.getElementById('data-list');
+Product.prototype.renderList = function () {
+  let button = document.getElementById("results");
+  let listItem = document.createElement("li");
+  listItem.innerText = `${this.name} was clicked ${this.clicks} times, and seen ${this.views} times.`;
+  button.appendChild(listItem);
+};
 
-  for(let i = 0; i < allProducts.length; i++) {
-    let product = allProducts[i];
-    let listItem = document.createElement('li');
-    listItem.innerText = `${product.name} was clicked ${product.clicks} times, and seen ${product.views} times.`;
-    divContainer.appendChild(listItem);
-  }
+function saveData() {
+  let stringifyImages = JSON.stringify(allProducts);
+  localStorage.setItem("images", stringifyImages);
```

Comment:
Work on your semantic naming. If someone else were to try to store a json file to local storage as "images" it would overwrite your data.